src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/BitbucketPullRequestDecorator.java

```diff
@@ -87,27 +86,44 @@ public String name() {
 
     @Override
     public DecorationResult decorateQualityGateStatus(AnalysisDetails analysisDetails, UnifyConfiguration configuration) {
+        String project = configuration.getRequiredProperty(PULL_REQUEST_BITBUCKET_PROJECT_KEY);
+        String repo = configuration.getRequiredProperty(PULL_REQUEST_BITBUCKET_REPOSITORY_SLUG);
+        String url = configuration.getRequiredProperty(PULL_REQUEST_BITBUCKET_URL);
+        String token = configuration.getRequiredProperty(PULL_REQUEST_BITBUCKET_TOKEN);
+        BitbucketConfiguration bitbucketConfiguration = new BitbucketConfiguration(url, token, repo, project);
+
+        client.set(BitbucketClientFactory.createClient(bitbucketConfiguration));
+
         try {
-            if(!client.supportsCodeInsights()) {
-                LOGGER.warn("Your Bitbucket instances does not support the Code Insights API.");
+            if (!client.get().supportsCodeInsights()) {
+                LOGGER.warn("Your Bitbucket instance does not support the Code Insights API.");
+                client.remove();
```

Comment:
Client simplification:
```suggestion
        BitbucketClient client = BitbucketClientFactory.createClient(bitbucketConfiguration);

        try {
            if (!client.suportsCodeInsights()) {
                LOGGER.warn("Your Bitbucket instance does not support the Code Insights API.");
```